apps/meteor/server/models/raw/Users.ts

```diff
@@ -0,0 +1,1231 @@
+					statusLivechat: 'available',
+					roles: 'livechat-agent',
+					...(ignoreAgentId && { _id: { $ne: ignoreAgentId } }),
+				},
+			},
+			{
+				$lookup: {
+					from: 'rocketchat_subscription',
+					let: { id: '$_id' },
+					pipeline: [
+						{
+							$match: {
+								$expr: {
+									$and: [
+										{ $eq: ['$u._id', '$$id'] },
+										{ $eq: ['$open', true] },
+										{ $ne: ['$onHold', true] },
+										{ ...(department && { $eq: ['$department', department] }) },
+									],
+								},
+							},
+						},
+					],
+					as: 'subs',
+				},
+			},
+			{
+				$lookup: {
+					from: 'rocketchat_livechat_department_agents',
+					localField: '_id',
+					foreignField: 'agentId',
+					as: 'departments',
+				},
+			},
+			{
+				$project: {
+					agentId: '$_id',
+					username: 1,
+					lastRoutingTime: 1,
+					departments: 1,
+					count: { $size: '$subs' },
+				},
+			},
+			{ $sort: { count: 1, lastRoutingTime: 1, username: 1 } },
+		];
+
+		if (department) {
+			aggregate.push({ $unwind: '$departments' });
+			aggregate.push({ $match: { 'departments.departmentId': department } });
+		}
+
+		aggregate.push({ $limit: 1 });
+
+		const [agent] = await this.col.aggregate<any>(aggregate).toArray();
+		if (agent) {
+			await this.setLastRoutingTime(agent.agentId);
+		}
+
+		return agent;
+	}
+
+	async getLastAvailableAgentRouted(department: string, ignoreAgentId: IUser['_id']) {
+		const aggregate: Array<Record<string, unknown>> = [
+			{
+				$match: {
+					status: { $exists: true, $ne: 'offline' },
+					statusLivechat: 'available',
+					roles: 'livechat-agent',
+					...(ignoreAgentId && { _id: { $ne: ignoreAgentId } }),
+				},
+			},
+			{
+				$lookup: {
+					from: 'rocketchat_livechat_department_agents',
+					localField: '_id',
+					foreignField: 'agentId',
+					as: 'departments',
+				},
+			},
+			{ $project: { agentId: '$_id', username: 1, lastRoutingTime: 1, departments: 1 } },
+			{ $sort: { lastRoutingTime: 1, username: 1 } },
+		];
+
+		if (department) {
+			aggregate.push({ $unwind: '$departments' });
+			aggregate.push({ $match: { 'departments.departmentId': department } });
+		}
+
+		aggregate.push({ $limit: 1 });
+
+		const [agent] = await this.col.aggregate<any>(aggregate).toArray();
+		if (agent) {
+			await this.setLastRoutingTime(agent.agentId);
+		}
+
+		return agent;
+	}
+
+	async setLastRoutingTime(userId: IUser['_id']) {
+		const result = await this.findOneAndUpdate(
+			{ _id: userId },
+			{
+				$set: {
+					lastRoutingTime: new Date(),
+				},
+			},
+			{ returnDocument: 'after' },
+		);
+		return result.value;
+	}
+
+	setLivechatStatusIf(userId: IUser['_id'], status: UserStatus, conditions: Filter<IUser> = {}, extraFields: Record<string, unknown> = {}) {
+		// TODO: Create class Agent
+		const query = {
+			_id: userId,
+			...conditions,
+		};
+
+		const update = {
+			$set: {
+				statusLivechat: status,
+				...extraFields,
+			},
+		};
+
+		return this.update(query, update);
```

Comment:
I believe it would be better to use `.updateOne`  for this and all the subsequent applicable cases. Or `updateMany` if that's the case instead of  `update` (it's deprecated)